tests/sanitize_defaults.cfg

```diff
@@ -13,4 +13,8 @@ replace: TIMESTAMP
 
 [Memory addresses]
 regex: (<[a-zA-Z_][0-9a-zA-Z_.]* at )(0x[0-9a-fA-F]+)(>)
-replace: \1MEMORY_ADDRESS\3
\ No newline at end of file
+replace: \1MEMORY_ADDRESS\3
+
+[Matplotlib figure size]
+regex: (Figure size )\d+x\d+( with \d+ Axes)
+replace: \1WIDTHxHEIGHT\2
```

Comment:
@bnavigator @bsipocz what is the purpose of `\1` and `\2` at the beginning and end of the replacement string (if you know)?

(And why is it `\2` at the end of the `WIDTHxHEIGHT` string but `\3` at the end of the `MEMORY_ADDRESS` string?)

Reply:
those are references for the regex groups. I haven't looked into the logic, but I expect that the aim is to ignore only the smallest, unpredictable part of the output while still failing the tests if e.g a figure with more Axes is generated than expected one.

As for groups, there are 3 of them in the memory addresses case and 2 in this one. I suppose the middle part, which is to be ignored, can be put in a group, but it shouldn't really matter in this case as there are no plans to reuse that group.

Reply:
Of course; very clever. Thank you!